pinot-spi/src/main/java/org/apache/pinot/spi/config/table/UpsertConfig.java

```diff
@@ -55,12 +55,16 @@ public enum Strategy {
   @JsonPropertyDescription("Column for upsert comparison, default to time column")
   private final String _comparisonColumn;
 
+  @JsonPropertyDescription("whether use snapshot for fast upsert metadata recovery")
+  private boolean _useSnapshot;
+
   @JsonCreator
   public UpsertConfig(@JsonProperty(value = "mode", required = true) Mode mode,
```

Comment:
We already got too many keys in this config. Suggest making a PR similar to #8743 to clean them up first.

Reply:
thanks @Jackie-Jiang. Will do it and rebase. 

one quick question, will this affect backward compatibility?

Reply:
We may deprecate the current constructor and add an empty constructor to keep backward compatibility

Reply:
updated in https://github.com/apache/pinot/pull/9150